src/cljs/main/broadfcui/main.cljs

```diff
@@ -78,40 +78,46 @@
                                                   :text [:span {} "FireCloud Forum" icons/external-link-icon]}]})]
           (when (= :registered (:registration-status @state))
             [header/GlobalSubmissionStatus])]]
-        (case (:registration-status @state)
-          nil [:div {:style {:margin "2em 0" :textAlign "center"}}
-               [comps/Spinner {:text "Loading user information..."}]]
-          :error [:div {:style {:margin "2em 0"}}
-                  (style/create-server-error-message (.-errorMessage this))]
-          :not-registered (profile-page/render
-                           {:new-registration? true
-                            :on-done #(do (nav/go-to-path :library)
-                                          (js-invoke (aget js/window "location") "reload"))})
-          :update-registered (profile-page/render
-                              {:update-registration? true
-                               :on-done #(do (nav/go-to-path :workspaces)
-                                             (js-invoke (aget js/window "location") "reload"))})
-          :registered
-          (if component
-            [component (make-props)]
-            [:h2 {} "Page not found."]))]))
+        (let [original-destination (aget js/window "location" "hash")
+              figure-out-destination (fn [fall-through]
```

Comment:
This figures out the destination, then loads the registration status. How about just calling it `on-done`? I would then be more inclined to expect multiple steps.